### PR TITLE
make PID_FILE work with nginx child processes

### PR DESCRIPTION
**Pull request by timcerexhe**

### CODE REVIEW

**Comment by jsuresh** [comment]
Did we decide that a better approach would be to look up the pid tree, until we found one that matches the pid file written (this has the nice property, that JCMS can actually write out the pid, and if it crashes without cleaning up nginx, we'll know about it).

@latos thoughts?

File: testapp/app.js [hunk 1]
```diff
@@ -54,8 +54,9 @@ var server = app.listen(appPort, function () {
 });
 
 function startProxy(frontPort, appPort) {
```
Comment (latos) [inline testapp/app.js]: Unrelated but can we rename "deploy2.json" to "deploy-with-nginx.json"

**Comment by latos** [comment]
@jsuresh I don't recall.  Either way I'd like to see some clearer documentation because I'm only barely following the approach now, and the mounting number of if-statements is making me uncomfortable without stupidly large amounts of tests.

**Comment by latos** [comment]
is this still a thing

**Pull request closed without merging**